db/dbformat.h

```diff
@@ -605,29 +592,36 @@ class IterKey {
   void EncodeLengthPrefixedKey(const Slice& key) {
     auto size = key.size();
     EnlargeBufferIfNeeded(size + static_cast<size_t>(VarintLength(size)));
-    char* ptr = EncodeVarint32(buf_, static_cast<uint32_t>(size));
+    char* ptr = EncodeVarint32(buf(), static_cast<uint32_t>(size));
     memcpy(ptr, key.data(), size);
-    key_ = buf_;
+    key_ = buf();
     is_user_key_ = true;
   }
 
   bool IsUserKey() const { return is_user_key_; }
 
  private:
-  char* buf_;
   const char* key_;
-  size_t key_size_;
-  size_t buf_size_;
-  char space_[32];  // Avoid allocation for short keys
-  bool is_user_key_;
+  size_t key_size_ : 32;
+  size_t buf_size_ : 31;
```

Comment:
```suggestion
  uint32_t key_size_;
  uint32_t buf_size_;
```

Reply:
The `size_t` here is intentional, because `uint32_t` produces warning like `shorten 64 bits to 32` :(